### PR TITLE
Update Cart.php

### PR DESCRIPTION
**Pull request by psandmore**
Default value for $id_product_attribute in deleteProduct method was changed from null to 0. Strict comparison $id_product_attribute !== null is not working with zero. Changed comparison with null to (int).

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       |  develop
| Description?  | any configuration
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | Does it break backward compatibility? no
| Deprecations? | Does it deprecate an existing feature? no
| Fixed ticket? | (optional) If this PR fixes an [Issue](https://github.com/PrestaShop/PrestaShop/issues), please write "Fixes #[issue number]" here.
| How to test?  | Please indicate how to best verify that this PR is correct.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15323)

### CODE REVIEW

File: classes/Cart.php [hunk 1]
```diff
@@ -1712,7 +1712,7 @@ public function deleteProduct(
                 WHERE `id_cart` = ' . (int) $this->id . '
                 AND `id_product` = ' . (int) $id_product . '
                 AND `id_customization` = ' . (int) $id_customization .
-                ($id_product_attribute != null ? ' AND `id_product_attribute` = ' . (int) $id_product_attribute : '')
+                (Int) $id_product_attribute ? ' AND `id_product_attribute` = ' . (int) $id_product_attribute : '')
```
Comment (matthieu-rolland) [inline classes/Cart.php]: thank you for your contribution @psandmore 

Could you please put the I of your "Int" in minuscule? 
There might also be unit tests to change due to your modification.

 thanx !

**Review by matthieu-rolland — CHANGES_REQUESTED** [review]

File: classes/Cart.php (hunk 1 above)
Comment (psandmore) [reply classes/Cart.php]: updated as requested

File: classes/Cart.php [hunk 2]
```diff
@@ -1712,7 +1712,7 @@ public function deleteProduct(
                 WHERE `id_cart` = ' . (int) $this->id . '
                 AND `id_product` = ' . (int) $id_product . '
                 AND `id_customization` = ' . (int) $id_customization .
-                ($id_product_attribute != null ? ' AND `id_product_attribute` = ' . (int) $id_product_attribute : '')
+                (int) $id_product_attribute ? ' AND `id_product_attribute` = ' . (int) $id_product_attribute : '')
```
Comment (matthieu-rolland) [inline classes/Cart.php]: there's a syntax error here, the last parenthesis line 1715 shouldn't be there maybe?
Dit it work for you?

**Review by matthieu-rolland — CHANGES_REQUESTED** [review]
fix syntax error

**Comment by PierreRambaud** [comment]
Hi and thanks for your contribution.
Can you please create an issue to explain what exactly happened et how we can reproduce.
Can you also complete the pull request description and title?

Kind regards

**Comment by psandmore** [comment]
I do not know how to reproduce. I saw in the code that the signature of the deleteProduct method changed. The $id_product_attribute default value was previously null and now it is zero. The original code used a strict comparison $id_product_attribute !== null. When the default value was null this was correct. Now the default value is zero and it is not working with zero. So I made a fix.

**Comment by matks** [comment]
Hi, [following our review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity following a request for modifications".

Since we did not get answer from author I close this PR.

Unfortunately, without a scenario "How to reproduce" we cannot test and validate this PR 😢 

Thank you for the contribution

**Pull request closed without merging**